### PR TITLE
Features/rails admin setup

### PR DESCRIPTION
**Pull request by seanculver**
This sets up rails admin on the production credits engine.
- Currently there is a conflict with the version of kaminari in rails_admin and sufia. This is currently being resolved in this PR. https://github.com/amatsuda/kaminari/pull/636
- All rails_admin config is moved into the production credits engine itself.
- Rails admin must be introspecting methods on models because a call to `production#name` blows up, temporary `try` used to get it up for now.
- When the PR is complete, we should make rails_admin a gemspec dependency.

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -1,6 +1,7 @@
 Rails.application.routes.draw do
   # Non Sufia generated routes
   mount ProductionCredits::Engine, at: "/production_credits"
+  mount RailsAdmin::Engine => '/production_credits/admin', as: 'rails_admin'
```
Comment (randycoulman) [inline config/routes.rb]: If we're moving RailsAdmin into the ProductionCredits engine, shouldn't that engine be responsible for mounting RailsAdmin instead?  Not sure if that makes sense or is even possible.

File: vendor/engines/production_credits/app/models/production_credits/production.rb [hunk 2]
```diff
@@ -12,7 +12,7 @@ def category_enum
     end
 
     def name
-      "#{production_name} - #{open_on.year}"
+      "#{production_name} - #{open_on.try(:year)}"
```
Comment (randycoulman) [inline vendor/engines/production_credits/app/models/production_credits/production.rb:15]: Do we really want a `nil` showing up wherever name is shown?  Maybe something like:

``` ruby
if open_on
  "#{production_name} - #{open_on.year}"
else
  "#{production_name} - unknown year"
end
```

File: config/routes.rb (hunk 1 above)
Comment (seanculver) [inline config/routes.rb]: oh interesting, yes I thought that had been removed. Good catch.

File: vendor/engines/production_credits/config/routes.rb [hunk 3]
```diff
@@ -1,4 +1,5 @@
 ProductionCredits::Engine.routes.draw do
   root to: 'homepage#index'
+  mount RailsAdmin::Engine => '/admin', as: 'rails_admin'
```
Comment (randycoulman) [inline vendor/engines/production_credits/config/routes.rb:3]: Wait, we are mounting it here.  Why do we need it in both places?

**Comment by randycoulman** [comment]
Done reviewing

File: vendor/engines/production_credits/app/models/production_credits/production.rb (hunk 2 above)
Comment (seanculver) [inline vendor/engines/production_credits/app/models/production_credits/production.rb:15]: Year is a required field, this fixes only RailsAdmin issues.

File: vendor/engines/production_credits/config/routes.rb (hunk 3 above)
Comment (seanculver) [inline vendor/engines/production_credits/config/routes.rb:3]: We don't, I forgot to hit save.

**Pull request merged**